### PR TITLE
Update PreviewController

### PR DESCRIPTION
**Pull request by rullzer**
The constructor is called with the userId. However if a user is not
logged in this is null. Which means that we get an exception instead of
this being handled gracefully in the middleware.

There are cleaner solutions. But this is the solution that is the
easiest to apply without lots of work and risk of breaking things
(handling the logged in middleware before initializing the controller
etc).

Basically this prevents events in sentry :wink:

### CODE REVIEW

**Review by danxuliu — COMMENTED** [review]
Should [the use of `$this->userId` in `getPreviewById()`](https://github.com/nextcloud/server/blob/e5cc8be9d5f8899417aa0716b8b29c596417af61/core/Controller/PreviewController.php#L138) be wrapped in a try/catch like [done in `getPreview()`](https://github.com/nextcloud/server/blob/e5cc8be9d5f8899417aa0716b8b29c596417af61/core/Controller/PreviewController.php#L103-L108)?

Should both of them also catch [`\OC\User\NoUserException`](https://github.com/nextcloud/server/blob/8fc47c6f002dd47518cea34ec5fe113fa5b915b7/lib/private/Files/Node/Root.php#L354) (which [is not documented in `IRootFolder`](https://github.com/nextcloud/server/blob/master/lib/public/Files/IRootFolder.php#L37-L44), but that is a different topic ;-) )?

**Comment by rullzer** [comment]
> Should [the use of `$this->userId` in `getPreviewById()`](https://github.com/nextcloud/server/blob/e5cc8be9d5f8899417aa0716b8b29c596417af61/core/Controller/PreviewController.php#L138) be wrapped in a try/catch like [done in `getPreview()`](https://github.com/nextcloud/server/blob/e5cc8be9d5f8899417aa0716b8b29c596417af61/core/Controller/PreviewController.php#L103-L108)?
> 
> Should both of them also catch [`\OC\User\NoUserException`](https://github.com/nextcloud/server/blob/8fc47c6f002dd47518cea34ec5fe113fa5b915b7/lib/private/Files/Node/Root.php#L354) (which [is not documented in `IRootFolder`](https://github.com/nextcloud/server/blob/master/lib/public/Files/IRootFolder.php#L37-L44), but that is a different topic ;-) )?

Well we know it is a user because we will check that during the middleware steps.

**Review by danxuliu — APPROVED** [review]
In @rullzer I trust :+1:

**Review by juliusknorr — APPROVED** [review]

**Review by icewind1991 — APPROVED** [review]
